scripts/firedrake-install

```diff
@@ -1006,6 +1006,114 @@ def build_and_install_slepc():
+                os.stat("/opt/cray")
+                log.info("This is a Cray system. Replacing cc/CC with gcc/g++")
+                # Set full path to gcc and g++
+                gcc_with_path = subprocess.run(["which", "gcc"], stdout=subprocess.PIPE, universal_newlines=True).stdout.rstrip()
+                gxx_with_path = subprocess.run(["which", "g++"], stdout=subprocess.PIPE, universal_newlines=True).stdout.rstrip()
+                cmake_args.extend(["-DCMAKE_C_COMPILER:FILEPATH=" + gcc_with_path,
+                                   "-DCMAKE_CXX_COMPILER:FILEPATH=" + gxx_with_path])
+                check_call(["cmake"] + cmake_args + ["../src/vtk"])
+                check_call(["cmake"] + cmake_args + ["../src/vtk"])
+            except OSError:
+                pass
+
+            # Always include these arguments
+            vtk_python_version = platform.python_version_tuple()[0]+'.'+platform.python_version_tuple()[1]
+            python_with_path = subprocess.run(["which", "python3"], stdout=subprocess.PIPE, universal_newlines=True).stdout.rstrip()
+            cmake_args.extend(["-DVTK_WRAP_PYTHON:BOOL=ON", "-DVTK_PYTHON_VERSION:STRING="+vtk_python_version,
+                               "-DPYTHON_EXECUTABLE:FILEPATH="+python_with_path])
+            check_call(["cmake"] + cmake_args + ["../src/vtk"])
+            cmake_args.extend(["-DVTK_ENABLE_KITS:BOOL=ON"])
+            check_call(["cmake"] + cmake_args + ["../src/vtk"])
+
+            # Disable neon optimisations if this is ARM64
+            machine_type = subprocess.run(["uname", "-m"], stdout=subprocess.PIPE, universal_newlines=True)
+            if machine_type.stdout.rstrip() == "aarch64":
+                log.info("This is an ARM64 machine. Disabling NEON optimisations")
+                cmake_args.extend(["-DCMAKE_C_FLAGS:STRING='-DPNG_ARM_NEON_OPT=0'",
+                                   "-DCMAKE_CXX_FLAGS:STRING='-DPNG_ARM_NEON_OPT=0'"])
+                check_call(["cmake"] + cmake_args + ["../src/vtk"])
+
+            # Build VTK
+            check_call(["make", "-j", "16"])
+
+            # Package VTK as a wheel
+        with directory("vtk/package"):
+            # Update setuptools and wheel
+            check_call(["pip3", "install", "--user", "--upgrade", "setuptools", "wheel"])
+
+            # Write a minimal README, setup.py and MANIFEST.in
+            with open('README', 'w') as f:
+                f.write("""
+VTK install for firedrake
+""")
+            with open('setup.py', 'w') as f:
+                f.write("""
+import setuptools
+
+setuptools.setup(
+    name="VTK",
+    version="8.2.0",
+    author="Kitware",
+    author_email="kitware@kitware.com",
+    description="The Visualization Toolkit (VTK)",
+    long_description="The Visualization Toolkit (VTK)",
+    long_description_content_type="text/markdown",
+    url="https://vtk.org",
+    packages=setuptools.find_packages(),
+    classifiers=[
+        "Operating System :: ARM64",
```

Comment:
Above we only check for non x86_64, but now we assume that this automatically means ARM. Is that really safe?